src/controllers/version/downloads.rs

```diff
@@ -89,9 +94,12 @@ pub async fn download(
                     // Non-canonical requests fallback to the "slow" path with a DB query, but
                     // we typically only get a few hundred non-canonical requests in a day anyway.
                     info_span!("cache.write", ?cache_key).in_scope(|| {
-                        app.version_id_cacher
-                            .blocking()
-                            .insert(cache_key, version_id)
+                        // SAFETY: This block_on should not panic. block_on will panic if the
+                        // current thread is an executor thread of a Tokio runtime. (Will panic
+                        // by "Cannot start a runtime from within a runtime"). Here, we are in
+                        // a spawn_blocking call because of conduit_compat, so our current thread
+                        // is not an executor of the runtime.
+                        rt.block_on(app.version_id_cacher.insert(cache_key, version_id))
```

Comment:
@tatsuya6502 is there a reason why you extracted the `rt` variable outside of the `conduit_compat()` closure instead of using `Handle::current.block_on(...)` directly here? we've been using `Handle::current.block_on(...)` in a few other places and I'm trying to figure out if that is a mistake or not 😅

Reply:
Hi. I did not realize that you already had `Handle::current().block_on(...)` in other places. I never tried doing so, and I am not sure if it is okay or not.

I had an impression that `Handle::current()` should be called in async context. But I just checked Tokio's document and it seems I was wrong. It says the followings:

https://docs.rs/tokio/1.33.0/tokio/runtime/struct.Handle.html#method.current

> `pub fn current() -> Self`
>
> Returns a `Handle` view over the currently running Runtime.
> 
> **Panics**
>
> This will panic if called outside the context of a Tokio runtime. That means that you must call this on one of the threads **being run by the runtime**, or from a thread with an active `EnterGuard`.

It says "outside the context of Tokio runtime", not "async context". `conduit_compat` runs the given closure in `tokio::task::spawn_blocking`, which uses a pool of threads **being run by Tokio runtime**. So, if understand correctly, calling `Handle::current()` should be okay there.

Reply:
okay, thanks for the clarification. I guess we'll try it out and see what happens. The test suite seems to work fine that way at least.